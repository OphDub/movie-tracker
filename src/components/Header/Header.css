@import url('https://fonts.googleapis.com/css?family=Fugaz+One');

header h1 {
  margin: 10px auto;
  font-family: 'Fugaz One', cursive;
  color: white;
  text-align: center;
  font-size: 3em;
}

header {
  background-image: url('../../assets/header-lights.jpg');
  background-size: cover;
  height: 130px;
  padding: 50px;
}