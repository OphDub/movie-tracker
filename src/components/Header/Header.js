import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux'

const Header = () => {
  return(
    <header>
      <h1>Movie Tracker</h1>
      <NavLink className="nav" to="/login">Login</NavLink>
      <NavLink to="/favorites"><button>Show Favorites</button></NavLink>
    </header>
  )
}

export default Header;