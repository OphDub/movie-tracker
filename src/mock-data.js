 export const cleanMovieArray = [
    {
      movie_id: 399055,
      title: 'The Shape of Water',
      vote_average: 7.4,
      vote_count: 794,
      overview: 'An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.',
      release_date: '2017-12-08',
      poster_path: 'https://image.tmdb.org/t/p/w500//iLYLADGA5oKGM92Ns1j9CDgk3iI.jpg'
    },
    {
      "movie_id": 284054,
      "title": "Black Panther",
      "vote_average": 5,
      "vote_count": 72,
      "overview": "T'Challa, the new ruler of the advanced kingdom of Wakanda, must defend his land from being torn apart by enemies from outside and inside the country.",
      "release_date": "2018-02-13",
      "poster_path": "https://image.tmdb.org/t/p/w500//bLBUCtMQGJclH36clliPLmljMys.jpg"
    }
] 

export const dirtyMovieObject = { results: [
    { 
      vote_average: 7.4,
      overview: 'An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.',
      vote_count: 794,
      title: 'The Shape of Water',
      poster_path: '/iLYLADGA5oKGM92Ns1j9CDgk3iI.jpg',
      release_date: '2017-12-08',
      id: 399055,
      randomnumber: 24341324142134,
    },
    {
      "title": "Black Panther",
      "vote_count": 72,
      "thing": "asdfasdfasdf",
      "id": 284054,
      "overview": "T'Challa, the new ruler of the advanced kingdom of Wakanda, must defend his land from being torn apart by enemies from outside and inside the country.",
      "release_date": "2018-02-13",
      "poster_path": "/bLBUCtMQGJclH36clliPLmljMys.jpg",
      "vote_average": 5,
    }
]}

export const oneMovie = {
  movie_id: 284054,
  title: 'Coco',
  vote_average: 7.7,
  vote_count: 1987,
  overview: 'Despite his familyâ€™s baffling generations-old ban on music, Miguel dreams of becoming an accomplished musician like his idol, Ernesto de la Cruz. Desperate to prove his talent, Miguel finds himself in the stunning and colorful Land of the Dead following a mysterious chain of events. Along the way, he meets charming trickster Hector, and together, they set off on an extraordinary journey to unlock the real story behind Miguel\'s family history.',
  posterImage: 'https://image.tmdb.org/t/p/w500//eKi8dIrr8voobbaGzDpe8w0PVbC.jpg'
}

export const userData = {
  id: 2,
  name: 'brophus park',
  email: 'bros@gmail.com',
  password: 'pass',
}

export const movieRawData = {
  data: {
      movie_id: 399055,
      title: 'The Shape of Water',
      vote_average: 7.4,
      vote_count: 794,
      overview: 'An other-worldly story, set against the backdrop of Cold War era America circa 1962, where a mute janitor working at a lab falls in love with an amphibious man being held captive there and devises a plan to help him escape.',
      release_date: '2017-12-08',
      poster_path: 'https://image.tmdb.org/t/p/w500//iLYLADGA5oKGM92Ns1j9CDgk3iI.jpg'
    }
}
