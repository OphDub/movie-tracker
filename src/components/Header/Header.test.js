/* eslint-disable */
import React from 'react';
import { Header, mapStateToProps, mapDispatchToProps } from './Header';
import { shallow } from 'enzyme';
import { userData } from '../../mock-data.js';

describe('Header', () => {
  let wrapper;
  
  beforeEach( () => {
    wrapper = shallow(
      <Header activeUser={userData} />);
  });

  it('should match the snapshot test', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps and mapDispatchToProps for Header', () => {
    it('should map the store correctly', () => {
      const mockUser = { id: 4, name: 'bruce', password: 'pass' };
      const mockStore = {
        activeUser: mockUser
      };

      const mapped = mapStateToProps(mockStore);

      expect(mapped.loginStatus).toEqual(mockStore.activeUser);
    });

    it('should call the dispatch func when sendFavorite is called', () => {
      const mockDispatch = jest.fn();
      const mapped = mapDispatchToProps(mockDispatch);

      mapped.logoutUser();

      expect(mockDispatch).toHaveBeenCalled();
    });
  });
});