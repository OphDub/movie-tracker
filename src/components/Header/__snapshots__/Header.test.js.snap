// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match the snapshot test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Movie Tracker
        </h1>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          className="nav"
          to="/login"
        >
          Login
        </NavLink>,
        <NavLink
          activeClassName="active"
          ariaCurrent="true"
          to="/favorites"
        >
          <button>
            Show Favorites
          </button>
        </NavLink>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Movie Tracker",
        },
        "ref": null,
        "rendered": "Movie Tracker",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": "Login",
          "className": "nav",
          "to": "/login",
        },
        "ref": null,
        "rendered": "Login",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "activeClassName": "active",
          "ariaCurrent": "true",
          "children": <button>
            Show Favorites
          </button>,
          "to": "/favorites",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Show Favorites",
          },
          "ref": null,
          "rendered": "Show Favorites",
          "type": "button",
        },
        "type": [Function],
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Movie Tracker
          </h1>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav"
            to="/login"
          >
            Login
          </NavLink>,
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            to="/favorites"
          >
            <button>
              Show Favorites
            </button>
          </NavLink>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Movie Tracker",
          },
          "ref": null,
          "rendered": "Movie Tracker",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": "Login",
            "className": "nav",
            "to": "/login",
          },
          "ref": null,
          "rendered": "Login",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": <button>
              Show Favorites
            </button>,
            "to": "/favorites",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Show Favorites",
            },
            "ref": null,
            "rendered": "Show Favorites",
            "type": "button",
          },
          "type": [Function],
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
